app/dashboard/views.py

```diff
@@ -1133,9 +1139,111 @@ def profile(request, handle):
                 },
             },
         }
+        return TemplateResponse(request, 'profiles/profile.html', context, status=status)
+
+    context['preferred_payout_address'] = profile.preferred_payout_address
+    order_by = request.GET.get('order_by', '-modified_on')
+    if context['preferred_payout_address']:
+        owned_kudos = Token.objects.filter(
+            owner_address=to_checksum_address(context['preferred_payout_address']),
+            contract__network=settings.KUDOS_NETWORK
+        ).order_by(order_by)
+        sent_kudos = Token.objects.filter(
+            kudos_transfer__from_address=to_checksum_address(context['preferred_payout_address']),
+            contract__network=settings.KUDOS_NETWORK
+        ).order_by(order_by)
+    else:
+        owned_kudos = None
+        sent_kudos = None
+
+    if owned_kudos:
+        owned_kudos_comments_public = []
+        for kudos_token in owned_kudos:
+            try:
+                owned_kudos_comments_public.append(kudos_token.kudos_transfer.comments_public)
+            except Exception as e:
+                logger.error(e)
+        logging.info(f'kudos comments_public: {owned_kudos_comments_public}')
+    else:
+        owned_kudos_comments_public = None
+    context['kudos_comments_public'] = owned_kudos_comments_public
+    context['kudos'] = owned_kudos
+    context['sent_kudos'] = sent_kudos
+
+    if request.method == 'POST' and request.is_ajax():
+        # Send kudos data when new preferred address
+        address = request.POST.get('address')
+        context['kudos'] = Token.objects.filter(owner_address=to_checksum_address(address)).order_by(order_by)
+        context['sent_kudos'] = Token.objects.filter(kudos_transfer__from_address=to_checksum_address(address)).order_by(order_by)
```

Comment:
We should `select_related` on the `'kudos_transfer'` since we're filtering on values from it here.